Code/Framework/AzCore/AzCore/RTTI/AttributeReader.h

```diff
@@ -165,27 +177,53 @@ namespace AZ
         template <class T, class U, typename... Args>
         bool Read(U& value, Args&&... args)
         {
-            return Internal::AttributeReader<T, U, Args...>::Read(value, m_attribute, m_instancePointer, AZStd::forward<Args>(args)...);
+            using DecayInstType = AZStd::remove_cvref_t<InstType>;
+            if constexpr (AZStd::is_pointer_v<DecayInstType> && AZStd::is_void_v<AZStd::remove_pointer_t<DecayInstType>>)
+            {
+                // Forward the void* as an rvalue so it is not a void*&
+                return Internal::AttributeReader<T, U, Args...>::Read(
+                    value, m_attribute, static_cast<DecayInstType>(m_instance), AZStd::forward<Args>(args)...);
+            }
+            else
+            {
+                return Internal::AttributeReader<T, U, Args...>::Read(value, m_attribute, m_instance, AZStd::forward<Args>(args)...);
+            }
         }
 
         template <typename RetType, typename... Args>
         bool Invoke(Args&&... args)
         {
-            return Internal::AttributeInvoke<RetType>(m_attribute, m_instancePointer, AZStd::forward<Args>(args)...);
+            using DecayInstType = AZStd::remove_cvref_t<InstType>;
+            if constexpr (AZStd::is_pointer_v<DecayInstType> && AZStd::is_void_v<AZStd::remove_pointer_t<DecayInstType>>)
+            {
+                // Forward the void* as an rvalue so it is not a void*&
+                return Internal::AttributeInvoke<RetType>(m_attribute, static_cast<DecayInstType>(m_instance), AZStd::forward<Args>(args...));
+            }
+            else
+            {
+                return Internal::AttributeInvoke<RetType>(m_attribute, m_instance, AZStd::forward<Args>(args...));
```

Comment:
These lines have the ellipsis in the wrong place:
```suggestion
                return Internal::AttributeInvoke<RetType>(m_attribute, static_cast<DecayInstType>(m_instance), AZStd::forward<Args>(args)...);
            }
            else
            {
                return Internal::AttributeInvoke<RetType>(m_attribute, m_instance, AZStd::forward<Args>(args)...);
```

Reply:
I'll fix this.